### PR TITLE
Do not try to transfer buddy

### PR DESCRIPTION
**Pull request by Gobberwart**
## Short Description:

Transfer attempt of buddy pokemon currently fails and results in endless transfer loop. Added check to both TransferPokemon and PokemonOptimizer to skip trying to transfer buddy pokemon.

## Fixes/Resolves/Closes (please use correct syntax):
- #5723

### CODE REVIEW

File: pokemongo_bot/cell_workers/transfer_pokemon.py [hunk 1]
```diff
@@ -31,7 +32,7 @@ def work(self):
 
         if self.bot.config.release.get('all'):
             group = [p for p in inventory.pokemons().all()
-                     if p.in_fort is False and p.is_favorite is False]
+                     if not p.in_fort and not p.is_favorite and not (p.unique_id == self.buddy['id'])]
```
Comment (DeXtroTip) [inline pokemongo_bot/cell_workers/transfer_pokemon.py]: If there is no buddy, `self.buddy['id']` will give an error

Comment (Gobberwart) [inline pokemongo_bot/cell_workers/transfer_pokemon.py]: Good point. Give me a few minutes.

**Comment by Gobberwart** [comment]
OK that should take care of buddy not set situation. @DeXtroTip, thoughts?

**Comment by DeXtroTip** [comment]
It could have been something like `p.unique_id == self.buddy.get('id', 0)`, i think. But yeah, it's okay now

**Comment by Gobberwart** [comment]
I thought about that, but figured it might still generate an error if NoneType isn't iterable. Then I thought about over-complicating that one line and just decided to palm it off to another method.

**Comment by Gobberwart** [comment]
If you're happy with this change @DeXtroTip could you please approve for merge?

**Pull request merged**